alerting/src/main/kotlin/org/opensearch/alerting/alerts/AlertIndices.kt

```diff
@@ -161,45 +206,74 @@ class AlertIndices(
 
         // if the indexes have been deleted they need to be reinitialized
         alertIndexInitialized = event.state().routingTable().hasIndex(ALERT_INDEX)
-        historyIndexInitialized = event.state().metadata().hasAlias(HISTORY_WRITE_INDEX)
+        alertHistoryIndexInitialized = event.state().metadata().hasAlias(ALERT_HISTORY_WRITE_INDEX)
+        findingHistoryIndexInitialized = event.state().metadata().hasAlias(FINDING_HISTORY_WRITE_INDEX)
+    }
+
+    private fun rescheduleAlertRollover() {
+        if (clusterService.state().nodes.isLocalNodeElectedMaster) {
+            scheduledRollover?.cancel()
+            scheduledRollover = threadPool
+                .scheduleWithFixedDelay({ rolloverAndDeleteAlertHistoryIndices() }, alertHistoryRolloverPeriod, executorName())
```

Comment:
What happens with the reschedule logic if the elected master has changed. Will the old master node continue to have the rollover activity scheduled?

Reply:
Yes it should still have the activity scheduled. I think future enhancements can be made to ensure 2 rollovers dont happen simultaneously.

Reply:
let's create an issue so that we address that as a follow up.